event-processing/cleanser/tests/unit/app.test.ts

```diff
@@ -0,0 +1,223 @@
+            event_name: "AUTHENTICATION_ATTEMPT",
+            component_id: "AUTH",
+            user
+        }
+
+        const outputMessage: ICleansedEvent = {
+            event_id: "123456789",
+            event_name: "AUTHENTICATION_ATTEMPT",
+            component_id: "AUTH",
+            timestamp: 1609462861,
+            timestamp_formatted: "2021-01-23T15:43:21.842"
+        }
+
+        const data : string = Buffer.from(TestHelper.encodeAuditEvent(outputMessage)).toString('base64')
+        const expectedResult : FirehoseTransformationResult = {
+            records: [{
+                data: data,
+                recordId: "7041e12f-c772-41e4-a05f-8bf25cc6f4bb",
+                result: "Ok"
+            }]
+        }
+
+        const firehoseEvent = TestHelper.createFirehoseEventWithEncodedMessage(TestHelper.encodeAuditEvent(exampleMessage));
+
+        const result = await handler(firehoseEvent);
+        expect(result).toEqual(expectedResult);
+    });
+
+    it('cleanses all messages when receiving an array', async () => {
+
+        const expectedData : string = Buffer.from(TestHelper.encodeAuditEventArray(TestHelper.exampleResultMessage)).toString('base64')
+        const expectedResult : FirehoseTransformationResult = {
+            records: [{
+                data: expectedData,
+                recordId: "7041e12f-c772-41e4-a05f-8bf25cc6f4bb",
+                result: "Ok"
+            }]
+        }
+
+        const firehoseEvent = TestHelper.createFirehoseEventWithEncodedMessage(TestHelper.encodeAuditEventArray(TestHelper.exampleMessage));
+
+        const result = await handler(firehoseEvent);
+
+        expect(result).toEqual(expectedResult);
+    });
+
+    it('retuns an empty cleansed event if the audit event does not have an event_id (impossible)', async () => {
+        const exampleMessage: IAuditEvent = {
+            client_id: "",
+            timestamp: 1609462861,
+            timestamp_formatted: "2021-01-23T15:43:21.842",
+            event_name: "AUTHENTICATION_ATTEMPT",
+            component_id: "AUTH"
+        }
+
+        const outputMessage: ICleansedEvent = {
+            event_id: "",
+            event_name: "",
+            component_id: "",
+            timestamp: 0,
+        }
+
+        const data : string = Buffer.from(TestHelper.encodeAuditEvent(outputMessage)).toString('base64')
+        const expectedResult : FirehoseTransformationResult = {
+            records: [{
+                data: data,
+                recordId: "7041e12f-c772-41e4-a05f-8bf25cc6f4bb",
+                result: "Ok"
+            }]
+        }
+
+        const firehoseEvent = TestHelper.createFirehoseEventWithEncodedMessage(TestHelper.encodeAuditEvent(exampleMessage));
+
+        const result = await handler(firehoseEvent);
+        expect(result).toEqual(expectedResult);
+    });
+
+    // it('obfuscates expected fields when receiving a single event using the secret string', async () => {

```

Comment:
If we dont need these can we remove the commented tests?

Reply:
Done.